x-pack/plugins/security_solution_serverless/public/get_started/card_step/index.tsx

```diff
@@ -12,78 +12,131 @@ import {
+  const { navigateTo } = useNavigateTo();
+
+  const stepContentRef = React.useRef<HTMLDivElement>(null);
+
   const isExpandedStep = expandedSteps.has(stepId);
+  const [panelClassNames, setPanelClassNames] = React.useState('');
   const steps = useMemo(
     () => getStepsByActiveProduct({ activeProducts, cardId, sectionId }),
     [activeProducts, cardId, sectionId]
   );
-  const { title, productLineRequired, description, splitPanel } =
+  const { title, description, splitPanel, icon, autoCheckIfStepCompleted } =
     steps?.find((step) => step.id === stepId) ?? {};
+  const hasStepContent = description != null || splitPanel != null;
+  const expandedStepPanelHeight = `calc(${stepContentRef.current?.offsetHeight}px + ${euiTheme.size.l} + ${euiTheme.size.xxxl})`;
+  const { indicesExist } = useStepContext();
+
+  useCheckStepCompleted({
+    autoCheckIfStepCompleted,
+    cardId,
+    indicesExist,
+    sectionId,
+    stepId,
+    toggleTaskCompleteStatus,
+  });
 
-  const badges = useMemo(() => getProductBadges(productLineRequired), [productLineRequired]);
+  const isDone = finishedSteps.has(stepId);
 
   const toggleStep = useCallback(
     (e) => {
       e.preventDefault();
-      const newState = !isExpandedStep;
-      onStepClicked({ stepId, cardId, sectionId, isExpanded: newState });
-    },
-    [cardId, isExpandedStep, onStepClicked, sectionId, stepId]
-  );
-
-  const isDone = finishedStepsByCard.has(stepId);
 
-  const hasStepContent = description != null || splitPanel != null;
+      if (hasStepContent) {
+        // Toggle step and sync the expanded card step to storage & reducer
+        onStepClicked({ stepId, cardId, sectionId, isExpanded: !isExpandedStep });
 
-  const handleStepButtonClicked = useCallback(
-    (e) => {
-      e.preventDefault();
-      onStepButtonClicked({ stepId, cardId, sectionId, undo: isDone ? true : false });
+        navigateTo({
+          deepLinkId: SecurityPageName.landing,
+          path: `#${stepId}`,
+        });
+      }
     },
-    [cardId, isDone, onStepButtonClicked, sectionId, stepId]
+    [hasStepContent, onStepClicked, stepId, cardId, sectionId, isExpandedStep, navigateTo]
   );
 
-  const panelClassNames = classnames({
-    'step-panel-collapsed': !isExpandedStep,
-    'step-panel-expanded': isExpandedStep,
-  });
+  useEffect(() => {
+    // Handle card expansion transition
+    if (isExpandedCard) {
+      setPanelClassNames(
+        classnames({
+          'step-panel-expanded': isExpandedCard,
+        })
+      );
+    } else {
+      setPanelClassNames(
+        classnames({
+          'step-panel-expanded': true,
+          'step-panel-collapsed': true,
+        })
+      );
+
+      setTimeout(() => {
+        setPanelClassNames(
+          classnames({
+            'step-panel-expanded': false,
+            'step-panel-collapsed': true,
+          })
+        );
+      }, HEIGHT_ANIMATION_DURATION);
+    }
+  }, [isExpandedCard, expandedSteps]);
```

Comment:
This is so odd, can't we manage transitions with CSS alone?

Reply:
I had a problem with the collapsing transition, it seems that if the `step-panel-expanded` removed as the same time `step-panel-collapsed` was added, the browser wouldn't detect the change and swallow the transition.
I was able to keep the collapsing transition by adding the `step-panel-collapsed` before removing `step-panel-expanded` so browser can detect the styling change.